File: eden/code-samples/eden.kt
/*
 * Eden is full of beautiful islands...
 */

class Island(
  val name: String,
  val length: Int,
  val width: Int,
  val features: Array<String>
) {
  val area: Int
    get() = this.calcArea()

  fun calcArea() =
    this.length * this.width 
}

// ...one of them is named Elysian

fun main() {
  Island(
    "Elysian",
    300,
    350,
    arrayOf("Sunsets", "Beaches")
  )
}

File: eden/code-samples/eden.rs
/* 
Eden is full of beautiful islands...
*/

#[derive(Debug)]
pub struct Island {
  name: String,
  length: u16,
  width: u16,
  features: Vec<String>,
}

impl Island {
  pub fn new(
    name: String,
    length: u16,
    width: u16,
    features: Vec<String>,
  ) -> Self {
    Island {
      name,
      length,
      width,
      features,
    }
  }

  pub fn area(&self) -> u16 {
    self.calc_area()
  }

  fn calc_area(&self) -> u16 {
    self.length * self.width
  }
}

/// ...one of them is named Elysian
fn main() {
  let elysian = Island::new(
    "Elysian".to_string(),
    300,
    350,
    vec!["Sunsets".to_string(), "Beaches".to_string()],
  );

  println!("{:?}", elysian);
}
